addons/web_editor/static/lib/odoo-editor/test/spec/copyPaste.test.js

```diff
@@ -1398,12 +1565,86 @@ describe('Copy and paste', () => {
+                        // select xxx in "<p>ab[xxx]cd</p>""
+                        const p = editor.editable.querySelector('p')
+                        let selection = {
+                            direction: Direction.FORWARD,
+                            anchorNode: p.childNodes[1],
+                            anchorOffset: 2,
+                            focusNode: p.childNodes[1],
+                            focusOffset: 5,
+                        }
+                        setTestSelection(selection, editor.document);
+                        editor._computeHistorySelection();
+                        // paste url
+                        await pasteText(editor, url);
+                        // Ensure the powerbox is active
+                        window.chai.expect(editor.commandBar._active).to.be.true;
+                        // Pick the second command (Paste as URL)
+                        triggerEvent(editor.editable, 'keydown', { key: 'ArrowDown' });
+                        triggerEvent(editor.editable, 'keydown', { key: 'Enter' });
                     },
-                    contentAfter: '<p>a<a href="http://existing.com">b<img src="https://download.odoocdn.com/icons/website/static/description/icon.png">[]c</a>d</p>',
+                    contentAfter: `<p>ab<a href="${url}">${url}</a>[]cd</p>`,
+                });
+            });
+            it('should restore selection after pasting image URL followed by UNDO', async () => {
+                const url = 'https://download.odoocdn.com/icons/website/static/description/icon.png';
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[abc]</p>',
+                    stepFunction: async editor => {
+                        await pasteText(editor, url);
+                        // Ensure the powerbox is active
+                        window.chai.expect(editor.commandBar._active).to.be.true;
+                        // Pick first command (Embed image)
+                        triggerEvent(editor.editable, 'keydown', { key: 'Enter' });
+                        // Undo
+                        await nextTick();
+                        editor.historyUndo();
+                    },
+                    contentAfter: '<p>[abc]</p>',
+                });
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[abc]</p>',
+                    stepFunction: async editor => {
+                        await pasteText(editor, url);
+                        // Ensure the powerbox is active
+                        window.chai.expect(editor.commandBar._active).to.be.true;
+                        // Pick second command (Paste as URL)
+                        triggerEvent(editor.editable, 'keydown', { key: 'ArrowDown' });
+                        triggerEvent(editor.editable, 'keydown', { key: 'Enter' });
+                        // Undo
+                        await nextTick();
```

Comment:
@dmo-odoo 
`Powerbox._currentValidade` (triggered by `keydown Enter`) was a synchronous function in 15.0, but it's async in 15.2...
Awaiting for the next tick makes sure the command's callback and powerbox's `postValidade` (which calls `historyStep`) have been run before calling `historyUndo`.